### ISSUE TITLE
E: The repository '.. ./ InRelease' is no longer signed.

### ISSUE THREAD

**Issue by a-t-0** [retorquere/zotero-deb#53]
Thank you for your work in presenting a packaged version of Zotero for Debian-based systems! While performing a `sudo apt update` I've noticed I'm getting an error that breaks my CI. The following output is provided:

```
Err:6 https://github.com/retorquere/zotero-deb/releases/download/apt-get ./ InRelease                  
  401  Unauthorized [IP: 185.199.111.133 443]
Reading package lists... Done                                                                      
E: Failed to fetch https://github.com/retorquere/zotero-deb/releases/download/apt-get/./InRelease  401  Unauthorized [IP: 185.199.111.133 443]
E: The repository 'https://github.com/retorquere/zotero-deb/releases/download/apt-get ./ InRelease' is no longer signed.
```
And I was wondering why this happens, and how I can resolve it without removing Zotero. Do you have any suggestions?

## System configuration
The output of `lsb_release -a`:
```
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 20.04.3 LTS
Release:	20.04
Codename:	focal
```
The Zotero version:
```
5.0.97.SOURCE.d530d9cf9
```

**Comment by retorquere** [comment]
Re-running the install.sh has no effect on the installation of Zotero, but un/reinstalling Zotero also has no effect on your Zotero data. Have you applied the `apt` fix?

**Comment by a-t-0** [comment]
To resolve the issue I have removed the ppa according to the instructions in [this link](https://linuxconfig.org/how-to-list-and-remove-ppa-repository-on-ubuntu-18-04-bionic-beaver-linux). I first tried the terminal command:
```
sudo add-apt-repository --remove ppa:retorquere/zotero
```
based on the output that was given by `apt policy`. That however did not remove the ppa. So I tried the GUI option by going to `Software & Updates` and removing it using the mouse. That removed the ppa, and resolved the issue.

Prior to this solution I also tried running `sudo apt update --fix-missing`, which did not resolve the issue. Thank you for your advice!

**Comment by retorquere** [comment]
Sorry about all of this man. The github chances have thrown everything in disarray. Things should be permanently better once zotero provides the hosting.

**Comment by juliangilbey** [comment]
Note for anyone else reading this issue: you have to run `apt update`, not `apt-get update` or `aptitude` to correctly change the repository to the new repository.

**Comment by retorquere** [comment]
That is not my experience. The tests running on this repository do not use `apt`, only `apt-get`.

**Comment by retorquere** [comment]
> To resolve the issue I have removed the ppa according to the instructions in [this link](https://linuxconfig.org/how-to-list-and-remove-ppa-repository-on-ubuntu-18-04-bionic-beaver-linux). I first tried the terminal command:
> 
> ```
> sudo add-apt-repository --remove ppa:retorquere/zotero
> ```

Nobody should be using this ppa. I thought I had taken this offline. 

> based on the output that was given by `apt policy`. That however did not remove the ppa. So I tried the GUI option by going to `Software & Updates` and removing it using the mouse. That removed the ppa, and resolved the issue.
> 
> Prior to this solution I also tried running `sudo apt update --fix-missing`, which did not resolve the issue. Thank you for your advice!

@juliangilbey were you responding to this in particular? I thought you meant switching between the github-release hosting an the sourceforge hosting.

**Comment by juliangilbey** [comment]
Yes indeed, and sorry for any confusion: my comment was intended to address how to switch between the different hosting / repositories and handle the aptitude errors.

**Comment by retorquere** [comment]
But were you trying to switch between ppa and the other (github/sourceforge) hosted repos, where you needed to use `apt`? Or between github and sourceforge?

I'll update the instructions to apt, just to make sure.

**Comment by retorquere** [comment]
Wait -- where did you find instructions using `apt-get`? The README already uses `apt`.

**Comment by a-t-0** [comment]
For completeness, before this thread escalates, I think the reason I stumbled upon this may be twofold:

0. I was not in complete control of a custom, [automated installation](https://github.com/HiveMinds/Productivity-setup/blob/main/src/apt_zotero.sh) process that I used as I was trying to include some plugins into the build whilst doing so, with limited knowledge on the matter.
1. Perhaps my installation procedure was performed quite a while ago, (before I became aware I should not use the ppa) (perhaps before you had taken the ppa offline), and it was only just recently that I ran the CI and stumbled upon the issue. 

So, my apologies for causing the distraction.
